### ISSUE TITLE
schemr List all scheme not working on windows(64) sublime 3 build 3114

### ISSUE THREAD

**Issue by trieditfirst** [benweier/Schemr#32]
I have installed latest sublime text 3 build: 3114.
I can change to next/previous scheme. Assign random screen
But unable to see list of all scheme on schemr with both keyboard shortcut and from ctrl+shift+p

Please let me know if I need to provide any details

reloading plugin CSS.css_completions
reloading plugin Diff.diff
reloading plugin HTML.encode_html_entities
reloading plugin HTML.html_completions
reloading plugin 0_package_control_loader.00-package_control
reloading plugin 0_package_control_loader.02-bz2
reloading plugin Origami.origami
reloading plugin Package Control.1_reloader
reloading plugin Package Control.2_bootstrap
reloading plugin Package Control.Package Control
reloading plugin Schemr.schemr
plugins loaded
Package Control: Skipping automatic upgrade, last run at 2016-05-25 15:08:21, next run at 2016-05-25 16:08:21 or after
Traceback (most recent call last):
  File "C:\Program Files\Sublime Text 3\sublime_plugin.py", line 553, in run_
    return self.run()
  File "schemr in C:\Users\me\AppData\Roaming\Sublime Text 3\Installed Packages\Schemr.sublime-package", line 247, in run
  File "schemr in C:\Users\me\AppData\Roaming\Sublime Text 3\Installed Packages\Schemr.sublime-package", line 99, in list_schemes
  File "schemr in C:\Users\me\AppData\Roaming\Sublime Text 3\Installed Packages\Schemr.sublime-package", line 205, in parse_scheme
NameError: global name 'a' is not defined

**Comment by objarni** [comment]
Same here, Ubuntu 15.10 64-bit ST3.

I looked at the source and len(a) is the source, whole row:

    			r, g, b = [background_color[i:i+2] for i in range(2, len(a), 2)]

.. inside the parse_scheme function. I cannot see 'a' anywhere there, however I'd guess it should be len(background_color) which is in the if some rows before...?

I don't know enough on the .packages format to actually change/test that though.

**Comment by objarni** [comment]
Hi, I just removed + installed Schemr again, and sadly another error pops up:

    Schemr.sublime-package", line 205, in parse_scheme
ValueError: need more than 2 values to unpack

**Comment by benweier** [comment]
I had this error occur when upgrading Schemr and it went away when I restarted Sublime.

**Comment by objarni** [comment]
@benweier I tried that just now, same error. Might there be some scheme cache that needs cleaning up?

**Comment by objarni** [comment]
Is there a debug-mode or something, that gives more verbose output of schemr?

**Comment by objarni** [comment]
Hi again, sorry to spam this thread, but I still got crashes. What I did:

1. I followed the cleanup-instructions  https://www.sublimetext.com/docs/3/revert.html
2. Installed Package Control (ST3)
3. Rebooted Sublime
4. Installed Schemr
5. Rebooted Sublime
6. Hit Alt+F5
7. Get crash.

    Sublime Text 3\Installed Packages\Schemr.sublime-package", line 205, in parse_scheme
ValueError: need more than 2 values to unpack

**Comment by objarni** [comment]
OK, I learned (by virtue of [1]) how to unpack installed packages, and modify files. I think I found the problem; the else clause of the background_color parser assumes the length is 8, when it is 6.

Modified if-else block:
```python
		if len(background_color) is 3:
			# Shorthand value, e.g. #111
			# Repeat the values for correct base 16 conversion.
			r, g, b = [background_color[i:i+1] * 2 for i in range(0, 3)]
		else:
			# Full-length color value, e.g. #111111 or #FFEEEEEE
			# Here we assume the order of hex values is #AARRGGBB
			# and so skip the first two chars
			print('Length: ', len(background_color))
			print('HEX: ', background_color)
			print('Range: ', list(range(2, len(background_color), 2)))
			r, g, b = [background_color[i:i+2] for i in range(2, len(background_color), 2)]

```

Output:
```
reloading plugin Schemr.schemr
Length:  6
HEX:  000000
Range:  [2, 4]
<< then exception occurs with "ValueError: need more than 2 values to unpack" >>
```

[1] https://forum.sublimetext.com/t/how-can-i-modify-built-in-packages-snippets/9939/2

**Comment by objarni** [comment]
Patch working for me:

```python
		try:
			if len(background_color) == 3:
				# Shorthand value, e.g. #111
				# Repeat the values for correct base 16 conversion.
				r, g, b = [background_color[i:i+1] * 2 for i in range(0, 3)]
			else:
				# Format #AARRGGBB or #RRGGBB
				start_index = 2 if len(background_color) == 8 else 0
				the_range = range(start_index, len(background_color), 2)
				r, g, b = [background_color[i:i+2] for i in the_range]
		except (Exception):
			print("Could not parse background_color: " + background_color)
			raise
```

**Comment by benweier** [comment]
I had originally tested on Windows 10/Sublime b3114 and OS X/Sublime b3113 with a small set of schemes but could not replicate the error. The issue should be fully resolved now that I've had a chance to test more thoroughly.

**Comment by objarni** [comment]
Ok cool. Strange that it used to assume format #AARRGGBB if the correct format is #RRGGBBAA - shouldn't that have been noticed before?

**Comment by objarni** [comment]
Fix works on Ubuntu 15.10 Sublime Text 3 b3114. Thanks for quick response!

**Comment by benweier** [comment]
Thanks for confirming @objarni. When I first set it to #AARRGGBB my research suggested this was the technically correct way to do it, but that's not how it turned out in reality.
